### PR TITLE
TechnologyCategoryControllerTest and CurriculumControllerTest completed

### PR DESCRIPTION
**Pull request by AmitCharran**
JavaDocs completed on testing as well

TechnologyCategoryController: Changed ObjectMapper.writeAsPrettyPrinter to ObjectMapper.WriteValueAsString
This change does not matter. Keep either one. I was just trying to figure out how to Mock the Post Methods.

Refactored TechnologyCategoryControllerTest to include test that gives forbidden when no user is logged in

Refactored CurriculumControllerTest to include test that gives forbidden when no user is logged in

### CODE REVIEW

File: src/main/java/com/revature/RevAssure/controller/TechnologyCategoryController.java [hunk 1]
```diff
@@ -60,8 +60,10 @@ public TechnologyCategoryController(TechnologyCategoryService technologyCategory
             if (revUser.isTrainer()) {
                 log.info("Trainer is creating a new technology category");
                 return ResponseEntity.ok().body(
-                        new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(
```
Comment (cptvictor) [inline src/main/java/com/revature/RevAssure/controller/TechnologyCategoryController.java:63]: why did you delete defaultPrettyPrinter()?

File: src/test/java/com/revature/RevAssure/controller/TechnologyCategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+     */
+    @WithMockUser
+    @Test
+    public void createTechnologyCategoryButForbiddenBecauseUserIsNotTrainerTest() throws Exception{
+        when(mockJwtUtil.extractUser(mockRevUserService)).thenReturn(mockUser);
+
+        mockMvc.perform(post("/technology_category")
+                .content(new ObjectMapper().writeValueAsString(technologyCategory))
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isForbidden())
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TechnologyCategoryControllerTest.java:149]: add .andExpect(jsonPath("$").doesNotExist());

File: src/test/java/com/revature/RevAssure/controller/TechnologyCategoryControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    /**
+     * Cannot add technology category because RevUser is a not logged in
+     * Http Status forbidden
+     */
+    @Test
+    public void createTechnologyCategoryButForbiddenBecauseUserIsNotLoggedIn403ForbiddenTest() throws Exception{
+        mockMvc.perform(post("/technology_category")
+                .content(new ObjectMapper().writeValueAsString(technologyCategory))
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isForbidden())
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TechnologyCategoryControllerTest.java:162]: add .andExpect(jsonPath("$").doesNotExist());

File: src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+    /**
+     * Cannot create topic. User is not logged in.
+     * HTTP Status forbidden
+     */
+    @Test
+    public void  createTopicTestButUserIsNotLogged403ForbiddenTest() throws Exception{
+        mockMvc.perform(post("/topic")
+                .content(new ObjectMapper().writeValueAsString(topicDTO))
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isForbidden())
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java:135]: add .andExpect(jsonPath("$").doesNotExist());

File: src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+     */
+    @WithMockUser
+    @Test
+    public void cannotCreateTopicBecauseATrainerIsNeededToCreateATopicTest() throws Exception{
+        when(mockJwtUtil.extractUser(mockRevUserService)).thenReturn(mockUser);
+
+        mockMvc.perform(post("/topic")
+                .content(new ObjectMapper().writeValueAsString(topicDTO))
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isForbidden())
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java:152]: add .andExpect(jsonPath("$").doesNotExist());

File: src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+
+    /**
+     * Cannot get topic by trainerId test
+     * HTTP Status forbidden
+     */
+    @Test
+    public void getAListOfTopicsByTrainerIdButNotLoggedIn403ForbiddenTest() throws Exception {
+        mockMvc.perform(get("/topic")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isForbidden())
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java:185]: add .andExpect(jsonPath("$").doesNotExist());

File: src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+    /**
+     * Cannot update topic. User is not logged in
+     * Http Status forbidden
+     */
+    @Test
+    public void UpdateTopicButNotLoggedIn403ForbiddenTest() throws Exception{
+        mockMvc.perform(put("/topic")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(new ObjectMapper().writeValueAsString(topicDTO)))
+                .andExpect(status().isForbidden())
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java:283]: add .andExpect(jsonPath("$").doesNotExist());

File: src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+     */
+    @WithMockUser
+    @Test
+    public void UpdateTopicTestButUserIsNotATrainer() throws Exception{
+        when(mockJwtUtil.extractUser(mockRevUserService)).thenReturn(mockUser);
+
+        mockMvc.perform(put("/topic")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(new ObjectMapper().writeValueAsString(topicDTO)))
+                .andExpect(status().isForbidden())
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java:299]: add .andExpect(jsonPath("$").doesNotExist());

File: src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,401 @@
...
+                .andReturn();
+    }
+
+    /**
+     * When a RevUser (Non-Trainer) requests topics by trainerId, will return and empty list
+     * HTTP Status ok
+     */
+    @WithMockUser
+    @Test
+    public void getAListOfTopicsByTrainerIdButARevAssociateRequestsItTest() throws Exception {
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java:195]: should empty lists be returned when user is associate or should it return 403 forbidden?

**Review by cptvictor — CHANGES_REQUESTED** [review]

**Review by DDRIV — DISMISSED** [review]
lgtm

File: src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,406 @@
...
+                .andReturn();
+    }
+
+    /**
+     * Update a topic
+     * Http Status Ok
+     */
+    @WithMockUser
+    @Test
+    public void UpdateTopicTest() throws Exception{
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/controller/TopicControllerTest.java:263]: lowercase u

**Review by cptvictor — CHANGES_REQUESTED** [review]

**Review by DDRIV — APPROVED** [review]
lgtm

**Review by cptvictor — APPROVED** [review]
lgtm
